NewForce-Cohort-5/tabloidfullstack-the-pollening
# Ticket #3 - View All Posts

# Description

This ticket focused on rendering a list of all posts so that a user can choose a post that seems interesting to read. When the user selects "Posts" from the navbar, they will be redirected to the posts page that will only display approved posts that have a publication date from the past. All posts will be in order by publication date with the most recent on top. Here is a breakdown of what was accomplished on both the client and server side:

Client-Side
-A /posts route path was added to ApplicationViews.js, along with wrapping that route with a PostProvider
-A /posts route was added to Header.js in the navbar as a navlink to view all posts
-Components were built out for Post.js and PostList.js
-Provider was built for PostProvider.js

Server-Side
-Post model, Post Repository, IPost Repository, and Post Controller were all built out to accomplish a successful api request to get all posts
-Ran into issues with pulling the id for a Post, but this error should be fixed now for this PR

## Type of change

Please delete options that are not relevant.


# Testing Instructions

As always, if you do not have seed data in your db, please run queries with both of these SQL scripts before testing this ticket:

[01_Tabloid_Create_DB.sql](https://github.com/NewForce-Cohort-5/tabloidfullstack-the-pollening/blob/main/SQL/01_Tabloid_Create_DB.sql)

[02_Tabloid_Seed_Data.sql](https://github.com/NewForce-Cohort-5/tabloidfullstack-the-pollening/blob/main/SQL/02_Tabloid_Seed_Data.sql)

After creating your seed data, use any of the emails currently in the UserProfile table to login. After a successful login, please navigate to /posts by selecting "Posts" in the navbar. You should now see a list of all posts set up in a React Bootstrap accordion. To view the author of the post and the category the post was written under, open the accordion to view both. A disabled button to read a post is shown here for a future ticket that involves viewing the details of post.

# Checklist:
